### ISSUE TITLE
Agent doesn't clean up status when agreement is canceled or even when agent is unregistered

### ISSUE THREAD

**Issue by dlarson04** [open-horizon/anax#2559]
```
root@plasmas1:~# hzn version
Horizon CLI version: 2.29.0-449
Horizon Agent version: 2.29.0-449
```

Steps to recreate:
- Register a node with BP to use hello-mms service (I used by deploy-hellomms)
- Allow agreement to form and service to start
- Unregister node
- Register node with different node policy to start a different service like vanilla helloworld (I used quay-perfhznmodel100.helloworld)
- Allow agreement to from and service to start
- Change node policy so that agreement is canceled and a new agreement isn't formed and no service running
- Issue `hzn exchange node liststatus <node>`

Expected result: running services field would be empty

Received result
```
root@plasmas1:~# hzn exchange node liststatus `hostname`_horizon0
{
  "services": [
    {
      "agreementId": "",
      "serviceUrl": "quay-perfhznmodel100.helloworld",
      "orgid": "ieam-roks-scale",
      "version": "1.0.0",
      "arch": "amd64",
      "containerStatus": [],
      "configState": "active"
    },
    {
      "agreementId": "",
      "serviceUrl": "deploy-hellomms",
      "orgid": "ieam-roks-scale",
      "version": "1.0.3",
      "arch": "amd64",
      "containerStatus": [],
      "configState": "active"
    }
  ],
  "runningServices": "|ieam-roks-scale/quay-perfhznmodel100.helloworld_1.0.0_amd64|ieam-roks-scale/deploy-hellomms_1.0.3_amd64|",
  "lastUpdated": "2021-06-09T13:08:25.031626Z[UTC]"
}
```

This causes the Edge UI to incorrectly show that this agent is running 2 services when it is running 0

![image](https://user-images.githubusercontent.com/8782353/121362330-06703080-c904-11eb-8ba1-8e081f98d194.png)

I think this is a P2 since it will result in the UI showing incorrect info

**Comment by Isabel-Gan** [comment]
Hello! I can work on this

**Comment by Isabel-Gan** [comment]
This issue will be fixed once #2267 is fixed. The terminated services are still showing up as active `MicroserviceDefinitions`, as microservices are not being updated correctly in the local database the status update function in `governance` is being given incorrect information on call to `FindNodeStatus`

**Comment by dlarson04** [comment]
verified in
root@plasmas1:~/scaler# hzn version
Horizon CLI version: 2.29.0-538
Horizon Agent version: 2.29.0-538
